core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider_life_cycle.go

```diff
@@ -148,26 +148,41 @@ func (p *logEventProvider) validateLogTriggerConfig(cfg LogTriggerConfig) error
 	if bytes.Equal(cfg.Topic0[:], zeroBytes[:]) {
 		return errors.New("invalid topic0: zeroed")
 	}
+	s := cfg.FilterSelector
+	if s >= 8 {
+		p.lggr.Error("filter selector %d is invalid", s)
+		return errors.New("invalid filter selector: larger or equal to 8")
+	}
 	return nil
 }
 
-// getFiltersBySelector the filters based on the filterSelector
-func (p *logEventProvider) getFiltersBySelector(filterSelector uint8, filters ...[]byte) []common.Hash {
-	var sigs []common.Hash
-	var zeroBytes [32]byte
-	for i, f := range filters {
-		// bitwise AND the filterSelector with the index to check if the filter is needed
-		mask := uint8(1 << uint8(i))
-		a := filterSelector & mask
-		if a == uint8(0) {
+// filterLogsByContent filters logs based on the filter selector and topics 1/2/3
+func (p *logEventProvider) filterLogsByContent(filter upkeepFilter, logs []logpoller.Log) []logpoller.Log {
+	if filter.selector == 0 {
+		return logs
+	}
+	var filteredLogs []logpoller.Log
+	// filter logs based on filter selector (topic1, topic2, topic3) if it's configured
+	checkTopic1 := filter.selector%2 == 1
+	checkTopic2 := (filter.selector>>1)%2 == 1
+	checkTopic3 := (filter.selector>>2)%2 == 1
+	for _, l := range logs {
+		if checkTopic1 && common.BytesToHash(l.Topics[1]).Hex() != filter.topics[1].Hex() {
+			p.lggr.Warnf("upkeep Id %s topics[1] %s does not match log topics[1] %s", filter.upkeepID, filter.topics[1].Hex(), common.BytesToHash(l.Topics[1]).Hex())
```

Comment:
nit. not sure if warn, nothing is wrong here, maybe debug?

Reply:
done